docs/subnets/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,979 @@
+Let's set it up. `Configure()` configures the `state` with the initial configuration at whatever `blockTimestamp` the precompile is enabled. In the HelloWorld example, we want to set up a default key value mapping in the state where the key is `storageKey` and the value is `Hello World!`. The `StateDB` allows us to store a key value mapping of 32 byte hashes. The below code snippet can be copied and pasted to overwrite the default `Configure()` code.
+
+```go
+// Configure configures [state] with the initial configuration.
+func (c *HelloWorldConfig) Configure(_ ChainConfig, state StateDB, _ BlockContext) {
+	// CUSTOM CODE STARTS HERE
+	// This will be called in the first block where HelloWorld stateful precompile is enabled.
+	// 1) If BlockTimestamp is nil, this will not be called
+	// 2) If BlockTimestamp is 0, this will be called while setting up the genesis block
+	// 3) If BlockTimestamp is 1000, this will be called while processing the first block
+	// whose timestamp is >= 1000
+	//
+	// Set the initial value under [common.BytesToHash([]byte("storageKey")] to "Hello World!"
+	res := common.LeftPadBytes([]byte("Hello World!"), common.HashLength)
+	state.SetState(HelloWorldAddress, common.BytesToHash([]byte("storageKey")), common.BytesToHash(res))
+}
+```
+
+#### Step 3.3 Modify Verify()
+
+We also see a `Verify()` function. `Verify()` is called on startup and an error is treated as fatal. We can leave this as is right now because there is no invalid configuration for the `HelloWorldConfig`. The below code snippet can be copied and pasted to overwrite the default `Verify()` code.
+
+```go
+// Verify tries to verify HelloWorldConfig and returns an error accordingly.
+func (c *HelloWorldConfig) Verify() error {
+	return nil
+}
+```
+
+#### Step 3.4 Modify sayHello()
+
+Next place to modify is in our `sayHello()` function. In a previous step we created the `IHelloWorld.sol` interface with two functions `sayHello()` and `setGreeting()`. We finally get to implement them here. If any contract calls these functions from the interface, the below function gets executed. This function is a simple getter function. In `Configure()` we set up a mapping with the key as `storageKey` and the value as `Hello World!` In this function, we will be returning whatever value is at `storageKey`. The below code snippet can be copied and pasted to overwrite the default `sayHello` code.
+
+```go
+func sayHello(accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	if remainingGas, err = deductGas(suppliedGas, SayHelloGasCost); err != nil {
+		return nil, 0, err
+	}
+	// No input provided for this function
+	// CUSTOM CODE STARTS HERE
+	// Get the current state
+	currentState := accessibleState.GetStateDB()
+	// Get the value set at recipient
+	value := currentState.GetState(HelloWorldAddress, common.BytesToHash([]byte("storageKey")))
```

Comment:
I didn't want to use constants here because I want people to have to copy and paste the least amount of code. If I define a constant then we have to copy and paste it at the top of the file and then this code as well.